lib/parser.js

```diff
@@ -192,52 +238,104 @@ function getValidator(version) {
   return validate;
 }
 
-async function customDocumentOperations(parsedJSON, asyncapiYAMLorJSON, initialFormat, options) {
+async function customDocumentOperations(
+  parsedJSON,
+  asyncapiYAMLorJSON,
+  initialFormat,
+  options
+) {
   validateServerVariables(parsedJSON, asyncapiYAMLorJSON, initialFormat);
-  validateServerSecurity(parsedJSON, asyncapiYAMLorJSON, initialFormat, SPECIAL_SECURITY_TYPES);
+  validateServerSecurity(
+    parsedJSON,
+    asyncapiYAMLorJSON,
+    initialFormat,
+    SPECIAL_SECURITY_TYPES
+  );
 
   if (!parsedJSON.channels) return;
 
   validateTags(parsedJSON, asyncapiYAMLorJSON, initialFormat);
   validateChannels(parsedJSON, asyncapiYAMLorJSON, initialFormat);
-  validateOperationId(parsedJSON, asyncapiYAMLorJSON, initialFormat, OPERATIONS);
-
-  await customComponentsMsgOperations(parsedJSON, asyncapiYAMLorJSON, initialFormat, options);
-  await customChannelsOperations(parsedJSON, asyncapiYAMLorJSON, initialFormat, options);
+  validateOperationId(
+    parsedJSON,
+    asyncapiYAMLorJSON,
+    initialFormat,
+    OPERATIONS
+  );
+
+  await customComponentsMsgOperations(
+    parsedJSON,
+    asyncapiYAMLorJSON,
+    initialFormat,
+    options
+  );
+  await customChannelsOperations(
+    parsedJSON,
+    asyncapiYAMLorJSON,
+    initialFormat,
+    options
+  );
 }
 
-async function validateAndConvertMessage(msg, originalAsyncAPIDocument, fileFormat, parsedAsyncAPIDocument, pathToPayload) {
+async function validateAndConvertMessage(
+  msg,
+  originalAsyncAPIDocument,
+  fileFormat,
+  parsedAsyncAPIDocument,
+  pathToPayload
+) {
   //check if the message has been parsed before
-  if (xParserMessageParsed in msg && msg[String(xParserMessageParsed)] === true) return;
-  const defaultSchemaFormat = getDefaultSchemaFormat(parsedAsyncAPIDocument.asyncapi);
+  if (xParserMessageParsed in msg && msg[String(xParserMessageParsed)] === true)
+    return;
+  const defaultSchemaFormat = getDefaultSchemaFormat(
+    parsedAsyncAPIDocument.asyncapi
+  );
   const schemaFormat = msg.schemaFormat || defaultSchemaFormat;
 
-  await PARSERS[String(schemaFormat)]({
+  const parser = PARSERS[String(schemaFormat)];
+
+  if (!parser) {
+    throw new ParserError({
```

Comment:
when using `groupValidationErrors`, do not throw error immediately. Let parser go through all messages, so they you throw error a the end, with all broken `schemaFormats` from all messages. Look into `customValidators` how they use it.

I think it will be trick, but yeah, try